projects/packages/jetpack-mu-wpcom/src/class-jetpack-mu-wpcom.php

```diff
@@ -521,4 +522,118 @@ public static function load_social_links() {
 			new \Automattic\Jetpack\Classic_Theme_Helper\Social_Links();
 		}
 	}
+
+	/**
+	 * Checks for translation updates and installs them if necessary.
+	 */
+	public static function maybe_update_translations() {
+		$plugin_slug = 'jetpack-mu-wpcom';
+
+		// Get installed site locales.
+		$locales = self::get_all_active_locales();
+		if ( empty( $locales ) ) {
+			return;
+		}
+
+		// Query the translation API.
+		$api_url      = 'https://translate.wordpress.com/api/translations-updates/wpcom/plugins';
+		$request_body = array(
+			'locales' => $locales,
+			'plugins' => array(
+				$plugin_slug => array( 'version' => 'latest' ),
+			),
+		);
+
+		$response = wp_remote_post(
+			$api_url,
+			array(
+				'body'    => wp_json_encode( $request_body ),
+				'headers' => array( 'Content-Type' => 'application/json' ),
+				'timeout' => 10,
+			)
+		);
+
+		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			return;
+		}
+
+		$data = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		if ( empty( $data['data'][ $plugin_slug ] ) ) {
+			return;
+		}
+
+		// Process each locale.
+		foreach ( $data['data'][ $plugin_slug ] as $translation ) {
+			$package_url   = $translation['package'] ?? '';
+			$locale        = $translation['wp_locale'] ?? '';
+			$last_modified = $translation['last_modified'] ?? '';
+
+			if ( ! $package_url || ! $locale || ! $last_modified ) {
+				continue;
+			}
+
+			// Get last saved timestamp for this locale.
+			$stored_last_modified = get_option( "jetpack_mu_wpcom_translation_{$locale}_last_modified", '' );
+
+			// If translation is already the latest version, skip.
+			if ( $stored_last_modified === $last_modified ) {
+				continue;
+			}
+
+			// Install the translation package.
+			if ( self::install_translation_package( $package_url ) ) {
+				update_option( "jetpack_mu_wpcom_translation_{$locale}_last_modified", $last_modified );
+			}
+		}
+	}
+
+	/**
+	 * Retrieves all active site locales.
+	 */
+	private static function get_all_active_locales() {
+		$locales             = array( get_locale() );
+		$available_languages = get_available_languages();
+		if ( ! empty( $available_languages ) ) {
+			$locales = array_unique( array_merge( $locales, $available_languages ) );
+		}
+		return array_values( $locales );
+	}
+
+	/**
+	 * Downloads and installs translation files from the given URL.
+	 *
+	 * @param string $package_url The URL of the language pack.
+	 *
+	 * @return bool True on success, false on failure.
+	 */
+	private static function install_translation_package( $package_url ) {
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+
+		$wpcomsh_dir = is_dir( WP_CONTENT_DIR . '/mu-plugins/wpcomsh-dev/' ) ? 'wpcomsh-dev' : 'wpcomsh';
```

Comment:
I note it's fairly likely that files placed in `mu-plugins/wpcomsh-dev/` will get deleted. As written, the code above probably won't restore them since it'll hit it "If translation is already the latest version, skip" check.

If it'll work, it might make more sense to write into `wp-content/languages/plugins/`, like WordPress does for translations it manages, or possibly `wp-content/languages/mu-plugins/` if you want to keep a plugin/mu-plugin separation.